src/technochat/chat/chat.go

```diff
@@ -80,19 +99,32 @@ func (c *Chat) SendAll(msg WSMessage) {
 func (c *Chat) HandleChatBroadcast() {
 	for {
 		select {
-		case <-time.After(ChatAFKLifetime * time.Second):
-			for id, _ := range c.corresps { //TODO: refactor
-				c.DelUser(id)
-			}
-			DelChat(c.ID)
+		case <-c.terminate:
 			return
 		case msg := <-c.broadcast:
-			for id, usr := range c.corresps {
-				err := usr.WS.WriteJSON(msg)
-				if err != nil {
-					log.Printf("error: chat: could not send a message to user %s: %v", usr.Name, err)
-					c.DelUser(id)
-				}
+			for _, usr := range c.corresps {
+				usr.send <- msg
```

Comment:
channel must be buffered and you must check if there is still space left, otherwise you may still block